src/visualBuilder/listeners/mouseHover.ts

```diff
@@ -79,36 +86,57 @@ async function addOutline(params?: AddOutlineParams): Promise<void> {
         fieldPath,
         fieldMetadata,
         fieldDisabled,
+        isLocked,
     } = params;
     if (!editableElement) return;
     const isVariant = !!fieldMetadata.variant;
-    addHoverOutline(editableElement as HTMLElement, fieldDisabled, isVariant);
     const fieldSchema = await FieldSchemaMap.getFieldSchema(
         content_type_uid,
         fieldPath
     );
-    if (!fieldSchema) return;
-    const { acl: entryAcl, workflowStage: entryWorkflowStageDetails, resolvedVariantPermissions } =
-        await fetchEntryPermissionsAndStageDetails({
-            entryUid: fieldMetadata.entry_uid,
-            contentTypeUid: fieldMetadata.content_type_uid,
-            locale: fieldMetadata.locale,
-            variantUid: fieldMetadata.variant,
-            fieldPathWithIndex: fieldMetadata.fieldPathWithIndex,
-        });
+    if (!fieldSchema) {
+        addHoverOutline(
+            editableElement as HTMLElement,
+            fieldDisabled,
+            isVariant,
+            isLocked
+        );
+        return;
+    }
+    const {
+        acl: entryAcl,
+        workflowStage: entryWorkflowStageDetails,
+        resolvedVariantPermissions,
+    } = await fetchEntryPermissionsAndStageDetails({
+        entryUid: fieldMetadata.entry_uid,
+        contentTypeUid: fieldMetadata.content_type_uid,
+        locale: fieldMetadata.locale,
+        variantUid: fieldMetadata.variant,
+        fieldPathWithIndex: fieldMetadata.fieldPathWithIndex,
+    });
     const { isDisabled } = isFieldDisabled(
         fieldSchema,
         eventDetails,
         resolvedVariantPermissions,
         entryAcl,
         entryWorkflowStageDetails
     );
-    addHoverOutline(editableElement, fieldDisabled || isDisabled, isVariant);
+    const currentIsLocked =
+        isLocked !== undefined
+            ? isLocked
+            : editableElement.getAttribute("data-field-locked") === "true";
```

Comment:
this logic is across the changes we can simplify this by doing

```suggestion
    const currentIsLocked = isLocked ?? editableElement.getAttribute("data-field-locked") === "true";
```